firmware/quadruna/BMS/src/app/app_soc.c

```diff
@@ -85,20 +103,26 @@ void app_soc_init(void)
     stats.is_corrupt = true;
     stats.charge_c   = -1;
 
+    uint8_t sd_read_data[64];
+
     // A negative soc value will indicate to app_soc_Create that saved SOC value is corrupted
-    // float saved_soc_c = -1.0f;
+    float saved_soc_c = -1.0f;
+
+#ifdef TARGET_EMBEDDED
 
     // TODO: Update to SD Card logic
     // if (app_eeprom_readSocAddress(&stats.soc_address) == EXIT_CODE_OK)
     // {
-    //     if (app_eeprom_readMinSoc(stats.soc_address, &saved_soc_c) == EXIT_CODE_OK)
-    //     {
-    //         if (IS_IN_RANGE(0.0f, SERIES_ELEMENT_FULL_CHARGE_C * 1.25f, saved_soc_c))
-    //         {
-    //             stats.charge_c   = (double)saved_soc_c;
-    //             stats.is_corrupt = false;
-    //         }
-    //     }
+    if (hw_sd_read(sd_read_data, DEFAULT_SOC_ADDR, 1) == SD_CARD_OK)
```

Comment:
Maybe add a comment about the 1 representing the number of blocks or make it a variable.